### ISSUE TITLE
Elementwise negative loglikelihood of VAE

### ISSUE THREAD

**Issue by delta2323** [chainer/chainer#1088]
In some situation, we need negative loglikelihood values for each output of the decoder of VAE. We already have `F.gaussian_nll` or `F.bernoulli_nll`, but they are loss functions, that is, they sum up these values and return the sum as a loss. We want a `Function`s that return each summand.

**Comment by cemoody** [comment]
This is true for most loss functions; instead of summing them you can do them elementwise and then sum. This is useful when you want to weight (or ignore) each example separately.

**Comment by delta2323** [comment]
One of the implementation choice is to add option (like `aggregate`, `sum`, `reduce` etc...) that enable/disable summation.

**Comment by ywatanabex** [comment]
I think there are three types of treatment over the 0-th axis (the axis of samples).
This is sometimes confusing for me.

1.  mean: take the average over the samples (e.g. `softmax_cross_entropy`)
2. sum: take the sum over the samples (e.g. `F.gaussian_nll`)
3. no aggregation: suggested in this issue. no summation over the samples

One approach is use `aggregation_method` option to select one of them.

**Comment by delta2323** [comment]
Taking mean along time axis can be problematic when we handle variable-length inputs.

**Comment by beam2d** [comment]
We now have `reduce` option for these functions.
